### PR TITLE
Support install kernel from tar ball and separate package split with comma

### PR DESCRIPTION
**Pull request by LiliDeng**
Need confirm for the URL format.
- Support tar and tar.gz format
- Recover /etc/resolv.conf file after do deprovision, details see [here ](https://github.com/Azure/WALinuxAgent/pull/1164)

### CODE REVIEW

File: Testscripts/Linux/customKernelInstall.sh [hunk 1]
```diff
@@ -302,22 +302,41 @@ function InstallKernel() {
             apt install -y wget
             apt remove -y linux-cloud-tools-common
             wget "$CustomKernel"
-            LogMsg "Installing ${CustomKernel##*/}"
-            dpkg -i "${CustomKernel##*/}"  >> $LOG_FILE 2>&1
-            kernelInstallStatus=$?
-            image_file=$(ls -1 *.deb* | grep -v "dbg" | sed -n 1p)
+            if [[ $CustomKernel = *.gz || $CustomKernel = *.tar ]]; then
+                tar -xf "${CustomKernel##*/}"
```
Comment (ader1990) [inline Testscripts/Linux/customKernelInstall.sh]: let s do it a little bit different, as it is a generic workflow.

If the kernel files are in a tar.gz or any kind of archive, just extract them before and then follow the same behaviour as before, if the archive contains debs, go for deb install and if it has rpms go for rpm installs. no need to duplicate code and so forth or to check if the url or path itself has deb in it

**Review by ader1990 — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/customKernelInstall.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/customKernelInstall.sh]: As @ader1990 said, if tar.gz and rpm has the same contents, deb files, please re-use the code.

**Comment by iamshital** [comment]
@LiliDeng Can you also add support for comma separated URLs?
-CustomKernel "https://xxxxxxxxxxx.x.x./kernel.deb,https://xxxxxxxxxxx.xxxxxxxxx.xxx/kernel-firmware.deb"

You can just download all of them in some unique temp folder, and install *.deb from that folder. No need to identify which one is the kernel package.

**Comment by chvalean** [comment]
> @LiliDeng Can you also add support for comma separated URLs?
> -CustomKernel "https://xxxxxxxxxxx.x.x./kernel.deb,https://xxxxxxxxxxx.xxxxxxxxx.xxx/kernel-firmware.deb"
> 
> You can just download all of them in some unique temp folder, and install *.deb from that folder. No need to identify which one is the kernel package.

For ubuntu azure kernel there is an install order required, tools requires common packages, and so on.
That remains valid for an archive as well.

There's some old code here - and it does not cover all  list of dependencies:
https://github.com/LIS/lis-test/blob/master/WS2012R2/lisa/remote-scripts/ica/ubuntu_test_kernel.sh#L122

**Comment by LiliDeng** [comment]
@chvalean Thanks!
dpkg -i *.deb
apt install -f -y this will fix dependency issue

**Comment by juhlee-microsoft** [comment]
@LiliDeng  any progress?

**Comment by LiliDeng** [comment]
@juhlee-microsoft Update based on @ader1990 comments and testing is ongoing, left tar ball for rpm packages, and work on Shital's requirement.

**Review by ader1990 — APPROVED** [review]

File: Testscripts/Linux/customKernelInstall.sh [hunk 2]
```diff
@@ -180,6 +180,40 @@ function Build_Kernel() {
...
+            CustomKernel="*.rpm"
+        fi
+        if [[ "${#array[@]}" -eq 1 ]]; then
+            CustomKernel="${CustomKernel##*/}"
+        fi
+    fi
+
+    if [[ $CustomKernel = *tar.gz || $CustomKernel = *.tar ]]; then
+        tar -xf "${CustomKernel#$LOCAL_FILE_PREFIX}"
+        count=$(find -name "*.deb" -type f | wc -l)
```
Comment (ader1990) [inline Testscripts/Linux/customKernelInstall.sh]: you can reuse this deb check also for the http part. just cycle trough the web files and download them or untar. then use this checks to see if it s rpm or deb. this way you remove 10 lines of code at least.

**Review by ader1990 — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/customKernelInstall.sh (hunk 2 above)
Comment (LiliDeng) [reply Testscripts/Linux/customKernelInstall.sh]: Thanks, for tar ball, also support localfile:./script/rpm/msft419.tar.gz this way, let me think about it.

**Comment by LiliDeng** [comment]
@ader1990 @chvalean @iamshital Any comments?

**Review by ader1990 — APPROVED** [review]
Looks good to me

**Review by chvalean — APPROVED** [review]

**Pull request merged**